app/waf/src/models/profiles/http_compress.model.ts

```diff
@@ -0,0 +1,376 @@
+        type: 'integer',
+        minimum: 1,
+        maximum: 128,
+      },
+    },
+    as3: {},
+  })
+  gzipWindowSize?: number;
+
+  @property({
+    type: 'boolean',
+    required: false,
+    default: false,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: true,
+      openapi: {
+        enum: [true, false],
+      },
+    },
+    as3: {},
+  })
+  keepAcceptEncoding?: boolean;
+
+  @property({
+    type: 'string',
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      openapi: {
+        minLength: 1,
+        maxLength: 100,
+      },
+    },
+    as3: {},
+  })
+  label?: string;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 1024,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 1024,
+      openapi: {
+        type: 'integer',
+        minimum: 128,
+        maximum: 131072,
+      },
+    },
+    as3: {},
+  })
+  minimumSize?: number;
+
+  @property({
+    type: 'string',
+    required: false,
+    default: 'gzip',
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 'gzip',
+      openapi: {
+        enum: ['gzip', 'deflate'],
+      },
+    },
+    as3: {},
+  })
+  preferMethod?: string;
+
+  @property({
+    type: 'string',
+    required: false,
+    default: '',
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 'any string',
+      openapi: {
+        minLength: 0,
+        maxLength: 4096,
+      },
+    },
+    as3: {},
+  })
+  remark?: string;
+
+  @property({
+    type: 'boolean',
+    required: false,
+    default: false,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: true,
+      openapi: {
+        enum: [true, false],
+      },
+    },
+    as3: {},
+  })
+  selective?: boolean;
+
+  @property({
+    type: 'array',
+    itemType: 'string',
+    required: false,
+    default: [],
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: ['http://example.org'],
+      openapi: {
+        items: {
+          type: 'string',
+          format: 'uri',
+        },
+      },
+    },
+    as3: {},
+  })
+  uriExcludes?: string;
+
+  @property({
+    type: 'array',
+    itemType: 'string',
+    required: false,
+    default: [],
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: ['http://example.org'],
+      openapi: {
+        items: {
+          type: 'string',
+          format: 'uri',
+        },
+      },
+    },
+    as3: {},
+  })
+  uriIncludes?: string;
+
+  @property({
+    type: 'boolean',
+    required: false,
+    default: true,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: true,
+      openapi: {
+        enum: [true, false],
+      },
+    },
+    as3: {},
+  })
+  varyHeader?: boolean;
```

Comment:
It should always have a value, so it does not need this '?'.

Reply:
removed.